### PR TITLE
use docker for selenium

### PR DESCRIPTION
**Pull request by ngaruko**
# Description

Use docker for selenium to avoid session creation issue due to version mismatch between the driver and the browser.

#7282 

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -466,14 +465,9 @@ module.exports = function(grunt) {
       'start-webdriver': {
         cmd:
           'mkdir -p tests/logs && ' +
-          './node_modules/.bin/webdriver-manager update && ' +
-          './node_modules/.bin/webdriver-manager start > tests/logs/webdriver.log & ' +
+          'docker run -d -p 4444:4444 -v /dev/shm:/dev/shm selenium/standalone-chrome:3.141.59-yttrium > tests/logs/webdriver.log & ' +
```
Comment (newtewt) [inline Gruntfile.js:468]: Looks like the [docs ](https://github.com/SeleniumHQ/docker-selenium#--shm-size2g) removed the volume mounting in favor of setting a cli arg. 

`docker run -d -p 4444:4444 --shm-size="2g" selenium/standalone-chrome:3.141.59-yttrium > tests/logs/webdriver.log &`

**Review by newtewt — CHANGES_REQUESTED** [review]
The docs read as if these are going to run in a headed browser because they've set it up to do so and then we set it to run headless in the `conf.js`. We normally run CI in headless and some people use headed browsers when debugging. Please add docs on how to run these so if someone is debugging they can run in a headed browser locally. Also, disable the "Xvfb" as described in the [docs ](https://github.com/SeleniumHQ/docker-selenium#running-in-headless-mode) for headless runs.  

Once we've migrated to WDIO this should no longer be an issue.

File: tests/constants.js [hunk 2]
```diff
@@ -4,14 +4,32 @@ const COUCH_HOST = 'localhost';
...
+  for (const net of nets[name]) {
+    // Skip over non-IPv4 and internal (i.e. 127.0.0.1) addresses
+    if (net.family === 'IPv4' && !net.internal) {
+      if (!results[name]) {
+        results[name] = [];
+      }
+      results[name].push(net.address);
+    }
+  }
+}
```
Comment (mrsarm) [inline tests/constants.js:22]: These nested `for`s:

```js
for (const name of Object.keys(nets)) {
  for (const net of nets[name]) {
```

Can be joined in one:

```js
for (const [name, net] of Object.entries(nets)) {
```

File: Gruntfile.js [hunk 3]
```diff
@@ -472,7 +472,9 @@ module.exports = function(grunt) {
       },
       'start-webdriver-ci': {
         cmd:
-          'scripts/e2e/start_webdriver.sh'
```
Comment (mrsarm) [inline Gruntfile.js:475]: Are we going to remove the `start_webdriver.sh` script? or are we keeping it because it's used/useful in other scenarios?

**Review by mrsarm — CHANGES_REQUESTED** [review]
Looks good, just a minor suggestion and a question.

**Comment by ngaruko** [comment]
No longer needed

**Pull request closed without merging**